api/views/phones.py

```diff
@@ -38,9 +38,26 @@
 )
 from ..serializers.phones import RealPhoneSerializer, RelayNumberSerializer
 
-
 logger = logging.getLogger("events")
 
+"""
+Twilio does not allow clients to delete messages that are "in-progress":
+
+https://www.twilio.com/docs/sms/api/message-resource#update-a-message-resource
+
+So, this custom response closes the connection first, which moves the message
+from "in-progress" to a finalized state. Then it calls the Twilio API to delete
+the message.
+
+See https://stackoverflow.com/a/4314182
+"""
+class DeleteTwilioMessageResponse(response.Response):
+    def close(self) -> None:
+        super().close()
+        sms_sid = self.renderer_context.get("request").data.get("SmsSid")
+        client = twilio_client()
+        client.messages(sms_sid).delete()
```

Comment:
I'm worried about what happens if something goes wrong here. I'm not sure what, if any error handling is done around the `close` method. It may depend on the WSGI server, and we can't really depend on the Django test client, which will probably have different WSGI behavior `./manage.py runserver`, which will be different than `gunicorn`.

On the other hand, a better way to handle this is to have an async task process delete the message some short time in the future. We'll probably need that async task worker this year, and can move that code then.

I'm OK shipping it and getting some practical experience in dev and stage with the error handling.

Alternatively, wrap the whole thing in an error handler to log the exception and send to Sentry


```suggestion
        try:
            sms_sid = self.renderer_context.get("request").data.get("SmsSid")
            client = twilio_client()
            client.messages(sms_sid).delete()
        except:
            logger.exception("DeleteTwilioMessageResponse failed.")
```